File: problems/560. Subarray Sum Equals K/solution.py
class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:
        count = sum = 0
        lookup = {0: 1}

        for num in nums:
            sum += num
            if (sum - k) in lookup:
                count += lookup[sum-k]
            if sum in lookup:
                lookup[sum] += 1
            else:
                lookup[sum] = 1

        return count

File: problems/560. Subarray Sum Equals K/solution.cpp
class Solution
{
public:
    int subarraySum(vector<int> &nums, int k)
    {
        int count(0), sum(0);
        unordered_map<int, int> hash({{0, 1}});
        for (int num : nums)
        {
            sum += num;
            if (hash.find(sum - k) != hash.end())
                count += hash[sum - k];
            hash[sum] += 1;
        }
        return count;
    }
};